src/graphnet/models/model.py

```diff
@@ -77,21 +72,60 @@ def fit(
             **trainer_kwargs,
         )
 
+    def fit(
+        self,
+        train_dataloader: DataLoader,
+        val_dataloader: Optional[DataLoader] = None,
+        *,
+        max_epochs: Optional[int] = 10,
+        gpus: Optional[Union[List[int], int]] = None,
+        callbacks: Optional[List[Callback]] = None,
+        ckpt_path: Optional[str] = None,
+        logger: Optional[Logger] = None,
+        log_every_n_steps: Optional[int] = 1,
+        gradient_clip_val: Optional[float] = None,
+        distribution_strategy: Optional[str] = "ddp",
+        **trainer_kwargs: Any,
+    ) -> None:
+        """Fit `Model` using `pytorch_lightning.Trainer`."""
+        self.train(mode=True)
+
+        self._construct_trainer(
+            max_epochs=max_epochs,
+            gpus=gpus,
+            callbacks=callbacks,
+            ckpt_path=ckpt_path,
+            logger=logger,
+            log_every_n_steps=log_every_n_steps,
+            gradient_clip_val=gradient_clip_val,
+            distribution_strategy=distribution_strategy,
+            **trainer_kwargs,
+        )
+
         try:
-            trainer.fit(
+            self._trainer.fit(
                 self, train_dataloader, val_dataloader, ckpt_path=ckpt_path
             )
         except KeyboardInterrupt:
             self.warning("[ctrl+c] Exiting gracefully.")
             pass
 
-    def predict(self, dataloader: DataLoader) -> List[Tensor]:
+    def predict(
+        self,
+        dataloader: DataLoader,
+        gpus: Optional[Union[List[int], int]] = None,
+    ) -> List[Tensor]:
         """Return predictions for `dataloader`.
 
         Returns a list of Tensors, one for each model output.
         """
         self.train(mode=False)
 
+        if not hasattr(self, "_inference_trainer"):
+            self._construct_trainer(
+                gpus=gpus,
+            )
+
```

Comment:
If this is _not_ the case, i.e., if `self._inference_trainer` has already been set, the `gpu` argument will be ignored. It might be worthwhile to inform the user in this case.
```suggestion
        elif gpus is not None:
            self.warning(
                "A `Trainer` instance has already been constructed, possibly "
                "when the model was trained. Will use this to get predictions. "
                f"Argument `gpus = {gpus}` will be ignored."
            )
```